synapse/lib/httpapi.py

```diff
@@ -99,7 +99,7 @@ def loadJsonMesg(self, byts):
         try:
             return json.loads(byts)
         except Exception:
-            self.sendRestErr('BadJson', 'Invalid JSON content.')
+            self.sendRestErr('SchemaViolation', 'Invalid JSON content.')
```

Comment:
This feels weird; since we're not schema validating.

Reply:
We are using a schema to validate the data though.  A SchemaViolation means "the data violates the schema" not "the schema is invalid" :)